play-zipkin-tracing/core/src/main/scala/jp/co/bizreach/trace/ZipkinTraceServiceLike.scala

```diff
@@ -57,14 +57,16 @@ trait ZipkinTraceServiceLike {
 
     Try(f) match {
       case Failure(t) =>
+        val timestamp = tracer.clock().currentTimeMicroseconds()
         Future {
           childSpan.tag("failed", s"Finished with exception: ${t.getMessage}")
-          childSpan.finish()
+          childSpan.finish(timestamp)
```

Comment:
> very low overhed

Oh, do you? finish() is executed in `Future` because I thought finish() has a little overhead unlike start(). But if not, it might be not required.

Meanwhile, `traceFuture` or `traceWS` method call finish() with `Future.onComplete` which need `ExecutionContext`. If the future of the function to trace has already been completed, a callback provided to `Future.onComplete` will either be applied immediately or be scheduled asynchronously. It is possible to be late the finish timestamp.

It might be a good idea to decide a policy. Do you have any thoughts?